src/wp-includes/rest-api/endpoints/class-wp-rest-sidebars-controller.php

```diff
@@ -125,9 +142,38 @@ public function get_items( $request ) {
 	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
 	 */
 	public function get_item_permissions_check( $request ) {
+		retrieve_widgets();
+
+		$sidebar = $this->get_sidebar( $request['id'] );
+
+		if ( ! $sidebar ) {
+			return new WP_Error( 'rest_sidebar_not_found', __( 'No sidebar exists with that id.' ), array( 'status' => 404 ) );
+		}
+
+		if ( $this->check_read_permission( $sidebar ) ) {
+			return true;
+		}
+
+
 		return $this->do_permissions_check();
 	}
 
+	/**
+	 * Checks if a sidebar can be read.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param array $sidebar Sidebar.
+	 * @return bool Whether the side can be read.
+	 */
+	public function check_read_permission( $sidebar ) {
+		if ( ! isset( $sidebar['show_in_rest'] ) || ! $sidebar['show_in_rest'] ){
+			return false;
+		}
+
+		return true;
```

Comment:
```suggestion
		return ! empty( $sidebar['show_in_rest'] );
```

This can be simplified to.

Reply:
Don't we need to check it the key `show_in_rest` exists first?

Reply:
No, `empty` won't error out on an undefined field.